docs/release_note_guidelines.rst

```diff
@@ -0,0 +1,189 @@
+    purposes, or to add extra metrics without having to make code changes and
+    re-deploy the service. See https://ddtrace.readthedocs.io/en/1.6.0/configuration.html
+    for more details.
+  features:
+  - |
+    dynamic instrumentation: Introduces the public interface for the dynamic instrumentation service. See
+    https://ddtrace.readthedocs.io/en/1.6.0/configuration.html for more details.
+
+Release Note Formatting
++++++++++++++++++++++++
+
+In general, a release note entry should follow the following format::
+
+    ---
+    <section>:
+      - |
+        scope: note
+
+Scope
+~~~~~
+
+This is a one-word scope shared between the release note and the conventional commit format used
+in the pull request title. This should the name of the library component, sub-component or integration
+that is impacted by this change. This should not be capitalized unless it is an acronym.
+
+To ensure consistency in component naming, the convention in referring to components is as follows:
+
+- Tracer: tracing
+- Profiler: profiling
+- Application security monitoring: ASM
+- Dynamic instrumentation: dynamic instrumentation
+- CI Visibility: CI visibility
+- Integrations: integration_name
+
+Note
+~~~~
+
+The note is a brief description of the change. It should follow sentence-case capitalization.
+The note should also follow valid restructured text (RST) formatting.
+
+- Features: Use present tense with the following format::
+
+  <scope>: This introduces <new_feature_or_component>.
+
+- Upgrade: Use present tense with the following formats::
+
+  <scope>: This upgrades <present tense explanation>.
+        With this upgrade, <actionable_step_for_user>
+  <scope>: <affected_code> has been removed. As an alternative to <affected_code>,
+        you can use <alternative> instead.
+
+- Deprecations: Use present tense for when deprecation/removal actually happens and future tense for
+when deprecation/removal is planned to happen. Include deprecation/removal timeline,
+as well as workarounds and alternatives in the following format::
+
+  <scope>: <affected_code> is deprecated and will be removed in <version_to_be_removed>.
+        As an alternative to <affected_code>, you can use <alternative> instead.
+
+- Fixes: Use past tense for the problem and present tense for the fix and solution in the following format::
+
+  <scope>: This fix resolves an issue where <ABC bug> caused <XYZ situation>.
+
+- Other: Since changes falling into this category are likely rare and not very similar to each other,
+no specific format other than a required scope is provided.
+The author is requested to use their best judgement to ensure a quality release note.
+
+- Prelude: Usually in tandem with a new feature or major change. No specific format other than a required scope
+is provided and again the author is requested to use their best judgement to provide context and background
+for the major change or new scope.
+
+Release Note Style Guidelines
```

Comment:
Should this section be near the top? These guidelines drive most of the other sections.